### ISSUE TITLE
Using the color utilities standalone.

### ISSUE THREAD

**Issue by sorahn** [Compass/compass#1847]
Hey guys, I'm using some compass parts, and since I don't have a lot of control over the import paths, It'd be nice if some of the files didn't require each other with such absolute pathing.

Example.

utilities/color just imports compass/utilities/color/contrast, but contrast also imports 'compass/utilities/color/brightness'.

My suggestion is to let utilities/colors handle them, and maybe just put warnings.  contrast could check for $brightness-is-imported and throw an error that way or something if you wanted to do it like that.

**Comment by scottdavis** [comment]
thats because some of the contrast mixins depend on brightness? I'm not sure what the problem you are describing is here?

**Comment by sorahn** [comment]
Yes, it does.

As I explained above I have limited ability (in our current envoronment) to set include directories for sass, and so our best option is to just checkout compass directly from github.  (using libsass, not ruby, sorry)

The problem is my folder structure looks like this now:
`/(repo_root)/bower_components/compass/core/stylesheets/compass/utilities/color/(contrast|brightness).scss`

And my import statement looks very similar:  
`@import "bower_components/compass/core/stylesheets/compass/utilities/color/contrast"`

But this fails because contrast tries to import brightness like so:  
`@import "compass/utilities/color/brightness"`

You also have a [this file](https://github.com/Compass/compass/blob/stable/core/stylesheets/compass/utilities/_color.scss) which does basically nothing.

My suggestion is that `utilities/_color.scss` contain the imports for both, however it's required for the library to use it, but no individual components import other ones.  Or at least, they should use relative pathing. `@import "brightness"`  or even `../../some/other/require`would work just fine.

**Comment by scottdavis** [comment]
I am not sure that would make much sense there is a clear dependency on brightness in contrast if we move it up to colors and someone just includes contrast it would not have the needed dependency to execute.

https://github.com/Compass/compass/blob/stable/core/stylesheets/compass/utilities/color/_contrast.scss#L29-L31

I fear this is just a current limitation of `libsass` and I don't think it's appropriate for compass to modify its behavior.

This seems like a edge case problem with your development environment.

Can't you set a load path for libsass via environment variables? /cc @hcatlin

**Comment by sorahn** [comment]
Yeah, moving env variables works well enough on my machine, but on the one we're going to use to build in production it's not as simple.

> if we move it up to colors and someone just includes contrast it would not have the needed dependency to execute.

I had a solution to this.

```
brightness.scss
$brightness-included: true;

contrast.scss
@if $brightness-included {
  cool
} @else {
  WARNING!
}
```

What about my last question.  And just having contrast require brightness via `@import "brightness";` instead of `@import "compass/utilities/color/brightness";`

> This seems like a edge case problem with your development environment.

Production environment actually, and yes, I believe my specific issue is an edge case, but I believe the practices that you guys are using makes it slightly harder to just piecemeal the parts together that you want.

My workaround solution is to maintain fork of compass that basically moves contents of the `/stylesheets` directory to the root.  Which works.

**Comment by scottdavis** [comment]
You are welcome to make a pull request and we can discuss it there further... but I'm not convinced the proposed solution scales well.

**Comment by sorahn** [comment]
I mean, the pull request would be...  

**_contrast.scss**

``` diff
--@import "compass/utilities/color/brightness"
++@import "brightness"
```

There's not too much to talk about.  Would you guys go for that?  Thats the hands down simplest solution.  You ask for 'contrast' and it looks for a file in the _same directory_ called 'brightness, rather than trying to start over from the import roots.

It's also worth pointing out that you guys switch between relative and root paths _all over the codebase_.  It's probably worth auditing the codebase and cleaning up _all_ the import statements.
